File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/brent/meduza/MeduzaApi.java
package com.github.bumblebee.command.brent.meduza;

import com.github.bumblebee.command.brent.meduza.response.MeduzaStockResponse;
import feign.RequestLine;

public interface MeduzaApi {

    String API_ROOT = "https://meduza.io";

    @RequestLine("GET /api/v3/stock/all")
    MeduzaStockResponse queryStock();
}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/brent/meduza/MeduzaApi.kt
package com.github.bumblebee.command.brent.meduza

import feign.RequestLine

data class MeduzaStockResponse(val usd: MeduzaStock, val eur: MeduzaStock, val brent: MeduzaStock)
data class MeduzaStock(val prev: Float, val current: Float, val state: String)

interface MeduzaApi {

    @RequestLine("GET /api/v3/stock/all")
    fun queryStock(): MeduzaStockResponse

    companion object {
        val API_ROOT = "https://meduza.io"
    }
}
